internal/auth/oidc/repository_auth_method_create.go

```diff
@@ -32,11 +33,19 @@ func (r *Repository) CreateAuthMethod(ctx context.Context, am *AuthMethod, _ ...
 		return nil, err // validate properly sets the op to the caller, the code and the msg, so just return it.
 	}
 
-	id, err := newAuthMethodId()
-	if err != nil {
-		return nil, errors.Wrap(err, op)
+	opts := getOpts(opt...)
+	am.PublicId = opts.withPublicId
+	if am.PublicId == "" {
+		id, err := newAuthMethodId()
+		if err != nil {
+			return nil, errors.Wrap(err, op)
+		}
+		am.PublicId = id
+	} else {
+		if !handlers.ValidId(handlers.Id(am.PublicId), AuthMethodPrefix) {
```

Comment:
this violates the layers of our system... you're calling an upper layer from the repo.  we need to refactor this func into a more util type thing or just dup the code here.

Reply:
Done.